libs/core/src/lib/calendar/calendar.component.ts

```diff
@@ -206,18 +208,51 @@ export class CalendarComponent<D> implements OnInit, ControlValueAccessor, Valid
     @ViewChild(CalendarHeaderComponent)
     _calendarHeaderComponent: CalendarHeaderComponent<D>;
 
+    /**
+     * Function used to disable previous button in the calendar header.
+     * @param date selected date
+     * @param currentlyDisplayedDate currently displayed date
+     * @param activeView current view of calendar
+     */
+    @Input()
+    previousButtonDisableFunction: NavigationButtonDisableFunction<D>;
+
+    /**
+     * Function used to disable next button in the calendar header.
+     * @param date selected date
+     * @param currentlyDisplayedDate currently displayed date
+     * @param activeView current view of calendar
+     */
+    @Input()
+    nextButtonDisableFunction: NavigationButtonDisableFunction<D>;
+
     /**
      * @hidden
      * Currently displayed days depending on month and year
      */
     _currentlyDisplayed: CalendarCurrent;
 
+    /**
+     * @hidden
+     */
+    get currentlyDisplayed(): CalendarCurrent {
+        return this._currentlyDisplayed;
+    }
+
+    set currentlyDisplayed(currentlyDisplayed: CalendarCurrent) {
+        this._currentlyDisplayed = currentlyDisplayed;
+    }
+
```

Comment:
in the recent changes you removed all logic from the setter. at this point we can work with `_currentlyDisplayed` directly as it is in main